R/sjTabItemAnalysis.R

```diff
@@ -372,14 +370,23 @@ sjt.itemanalysis <- function(df,
   # -------------------------------------
   # check if html-content should be printed
   # -------------------------------------
-  out.html.table(no.output, file, knitr, complete.page, use.viewer)  
-  invisible(list(class = c("sjTable", "sjtitemanalysis"),
-                 df.list = df.ia,
-                 index.scores = index.scores,
-                 df.index.scores = df.index.scores,
-                 cronbach.values = cronbach.total,
-                 ideal.item.diff = diff.ideal.list,
-                 knitr = knitr,
-                 knitr.list = knitr.list,
-                 complete.page = complete.page))
+  #out.html.table(no.output, file, knitr, complete.page, use.viewer)  
+  
+  structure(
+    class = c("sjTable", "sjtitemanalysis"),
+    list(
+      output.complete = complete.page,
+      knitr = knitr,
+      file = file,
+      show = !no.output,
+      use.viewer = use.viewer,
+      df.list = df.ia,
+      index.scores = index.scores,
+      df.index.scores = df.index.scores,
+      cronbach.values = cronbach.total,
+      ideal.item.diff = diff.ideal.list,
+      knitr.list = knitr.list
```

Comment:
Ok, knitr-list are just the single elements of the complete output - so, it's not needed in the S3-print-method.